### PR TITLE
Significant tidy up of the abundances module and notebook

### LINKED ISSUE #144: Look at providing scalings for individual abundances
N/O appears to be a function of Z (i.e. N does not uniformly scale with Z). We might want an elegant way to define additional scalings.

### PR DESCRIPTION
**Pull request by stephenmwilkins**
This is a significant tidy-up of the abundances module and supersedes a now closed PR.

- I have entirely changed how depletion is implemented to better align with Cloudy 23. Depletion can be accounted for either providing a depletion pattern in the form of a dictionary or a depletion_method object. When a depletion_method is provided there is an optional scaling factor that can be used.
- This has accommodated the inclusion of the Jenkins (2009) model which is now implemented in Cloudy '23.
- Additional Solar abundance patterns have been added for transparency and flexibility.
- Significant improvement to the (IMO already good) doc strings.
- The notebook docs have been expended to demonstrate this new functionality.

**This is best reviewed by following the notebook.**

Closes #144 

## Issue Type

- [] Bug
- [x] Document
- [x] Enhancement

## Checklist
- [x] I have read the [CONTRIBUTING.md]() -->
- [x] I have added docstrings to all methods
- [x] I have added sufficient comments to all lines
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no pep8 errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/synthesizer/abundances.py [hunk 1]
```diff
@@ -92,41 +98,33 @@ def C(metallicity):
...
-        "V",
-        "Cr",
-        "Mn",
-        "Fe",
-        "Co",
-        "Ni",
-        "Cu",
-        "Zn",
-    ]
+class ElementDefinitions:
```
[the remark this answers is not in the document]
Comment (stephenmwilkins) [reply src/synthesizer/abundances.py:101]: I'm on the fence. I think the alternative is that we probably have `abundances/` with individual modules for the element definitions, solar metallicity patterns and depletion patterns.

[the remark this answers is not in the document]
Comment (WillJRoper) [reply src/synthesizer/abundances.py:101]: That's exactly how it should be done. Nail on the head.

[the remark this answers is not in the document]
Comment (WillJRoper) [reply src/synthesizer/abundances.py:101]: Essentially your container classes should be submodules in a module (directory) called abundances.

**Review by WillJRoper — CHANGES_REQUESTED** [review]
As discussed in comments: 
- Create abundances module.
- Convert container classes into submodules of this module. (None of them have inits so the behaviour is identical with this structure just without capitilisation.)
- The subclasses are then just classes within the submodules.

Effectively you can just copy what you have inside each container into scripts with the name of the container (minus capital letter). 

Just a general note for the future: This subclass structure is only useful when you want to define a class which you want to protect so that it's only accessible if another class has been instantiated (e.g. A `Wheel` class only matters if you have a `Car` class.) Imporantly the outer class here has an init and is not just a container. An outer class with no init behaves just like a module (e.g. file) but leads to longer files.

File: src/synthesizer/abundances.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (stephenmwilkins) [reply src/synthesizer/abundances.py]: Implemented.

**Comment by stephenmwilkins** [comment]
> Great! Needs a quick removal of the unused import but could you rename abundances.py to abundance_patterns.py? Just to differentiate it from the module and follow the convention you've used for the solar_abundance_patterns submodule?

I guess, I suppose a user won't see that if we keep the stuff in __init__, i.e. as long as a user can do:

`from synthesizer.abundances import Abundances`

**Comment by WillJRoper** [comment]
Implemented. Just for sanities sake.

**Comment by WillJRoper** [comment]
The user will only ever see `from synthesizer.abundances import Abundances`

**Review by WillJRoper — APPROVED** [review]

**Pull request merged**